src/hotspot/share/classfile/moduleEntry.cpp

```diff
@@ -454,6 +454,7 @@ void ModuleEntry::init_as_archived_entry() {
   if (_location != NULL) {
     _location = ArchiveBuilder::get_relocated_symbol(_location);
   }
+  JFR_ONLY(memset(trace_id_addr(), 0, sizeof(traceid)));
```

Comment:
`memset` looks dodgy here. Maybe `JFR_ONLY(set_trace_id(0))`?

Reply:
I removed these from my previous commit, because the trace_id seemed to be predictable and wouldn't cause the archive content to change. Anyway, I've added the code you suggested in commit [5706a82](https://github.com/openjdk/jdk/pull/267/commits/5706a821a4b96c1c44c44ccf6d78c0659a6f2976). That way, if trace_ids become unpredictable due to future JFR implementation changes, the CDS code won't be affected.